packages/query/src/query-engine/index.ts

```diff
@@ -169,94 +80,18 @@ export class QueryEngine {
 		scriptModules: ScriptModuleProps[] | undefined;
 		bodyClasses: string[] | undefined;
 	} > => {
-		if ( ! QueryEngine.isClientInitialized ) {
-			QueryEngine.initialize();
-		}
 		const variables = { uri };
 
-		try {
-			const data = await QueryEngine.apolloClient.query( {
-				query: GetCurrentTemplateDocument,
+		const data = await fetchQuery( {
+			name: [ 'templateData', uri ],
```

Comment:
Using variables for additional cache keys is an adapter-level concern.
```suggestion
			name: 'GetCurrentTemplate'
```

Reply:
https://github.com/rtCamp/snapwp/pull/144#discussion_r2053498679

Tanstack uses this string as a cache key, so here we are caching each page individually by having uri as part of the cache key

Reply:
If (_big_ if) that's a valid implementation concern, then it's a _tanstack_ concern, so _tanstack_ should be responsible for extracting _any_ variables into their relevant cache keys.

Reply:
It's not about query variable, tanstack simply uses array of key as a cache key, so We want individual cache for each page then we pass the uri which will be dynamic at the same time allows to create unique key for each page so they can be cached properly.